From: Kill Your TV <killyourtv@i2pmail.org>
Date: Sun, 12 Oct 2014 01:45:33 +0000
Subject: fix pidfile location

When running as a daemon, I want the PID to be located under /run to make
interaction with the initscript and systemd less convoluted.

---
 DaemonLinux.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/DaemonLinux.cpp b/DaemonLinux.cpp
index 530eb07..3f828d3 100644
--- a/DaemonLinux.cpp
+++ b/DaemonLinux.cpp
@@ -74,8 +74,7 @@ namespace i2p
 			}
 
 			// Pidfile
-			pidfile = i2p::util::filesystem::GetDataDir().string();
-			pidfile.append("/i2pd.pid");
+			pidfile = "/run/i2pd/i2pd.pid";
 			pidFilehandle = open(pidfile.c_str(), O_RDWR | O_CREAT, 0600);
 			if (pidFilehandle == -1)
 			{
